src/basic_types.hpp

```diff
@@ -38,10 +38,9 @@ enum class AmrTag : int { derefine = -1, same = 0, refine = 1 };
 struct SimTime {
   SimTime() = default;
   SimTime(const Real tstart, const Real tstop, const int nmax, const int ncurr,
-          const int nout)
-      : start_time(tstart), time(tstart), tlim(tstop),
-        dt(std::numeric_limits<Real>::max()), nlim(nmax), ncycle(ncurr),
-        ncycle_out(nout) {}
+          const int nout, const Real dt_in = std::numeric_limits<Real>::max())
```

Comment:
Why is `dt_in` set to maximum double by default? This seems a little strange to me... should it not be set to either  `inf` or 0? Or signal NaN?

Reply:
Which would you prefer?

Reply:
I think the safest choice is probably a signal `NaN` and that's where I lean. Though maybe @jdolence @jmstone or @pgrete has thoughts on this.

Reply:
Setting it to 0 doesn't work because for whatever reason dt is not initialized in the advection example, so it takes min(dt,whatever it computes), which turns out to be 0 and the simulation never progresses (i.e. t stays at t_start).

Setting it to signaling_nan causes the simulation to stop because all comparisons return as true.  

So, I've left it as max().